ldi-orchestrator/ldio-application/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/repositories/PipelineFileRepository.java

```diff
@@ -0,0 +1,152 @@
+	private final ObjectWriter writer = mapper.writer(new DefaultPrettyPrinter());
+	private final ObjectReader reader = mapper.readerFor(PipelineConfig.class);
+
+	public PipelineFileRepository(OrchestratorConfig config) {
+		activePipelines = new HashMap<>();
+		directory = new File(config.getDirectory());
+		directory.mkdirs();
+	}
+
+	@Override
+	public List<PipelineConfigTO> findAll() {
+		return activePipelines.values().stream().map(SavedPipeline::pipelineConfig).toList();
+	}
+
+	public Map<File, PipelineConfigTO> getStoredPipelines() {
+		try (Stream<Path> files = Files.list(directory.toPath())) {
+			return files
+					.filter(path -> !Files.isDirectory(path))
+					.filter(path -> path.toFile().getName().endsWith(EXTENSION_YML)
+							|| path.toFile().getName().endsWith(EXTENSION_YAML))
+					.map(path -> {
+						try {
+							return Map.of(path.toFile(), readConfigFile(path));
+						} catch (PipelineParsingException e) {
+							log.error(e.getMessage());
+							return null;
+						}
+					})
+					.filter(Objects::nonNull)
+					.map(Map::entrySet)
+					.flatMap(Collection::stream)
+					.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+		} catch (IOException e) {
+			throw new UncheckedIOException(e);
+		}
+	}
+
+
+	@Override
+	public void save(PipelineConfig pipeline) {
+		if (exists(pipeline.getName())) {
+			throw new PipelineAlreadyExistsException(pipeline.getName());
+		}
+		var pipelineConfig = fromPipelineConfig(pipeline);
+		var savedFile = pipelineFile(pipeline.getName());
+
+		if (savedFile.exists()) {
+			AtomicInteger count = new AtomicInteger();
+			boolean success = false;
+			do {
+				savedFile = new File(directory, savedFile.getName().replace(EXTENSION_YML, ("(%d)%s").formatted(count.incrementAndGet(), EXTENSION_YML)));
+				if (!savedFile.exists()) {
+					success = true;
+				}
+			} while (!success);
+		}
+
+		try {
+			writer.writeValue(savedFile, pipelineConfig);
+			activePipelines.put(pipeline.getName(), new SavedPipeline(pipelineConfig, savedFile));
+		} catch (IOException e) {
+			throw new UncheckedIOException(e);
+		}
+	}
+
+	@Override
+	public void save(PipelineConfig pipeline, File persistedFile) {
+		if (exists(pipeline.getName())) {
+			throw new PipelineAlreadyExistsException(pipeline.getName());
+		}
+		activePipelines.put(pipeline.getName(), new SavedPipeline(fromPipelineConfig(pipeline), persistedFile));
+	}
+
+
+	@Override
+	public void delete(String pipelineName) {
+		ofNullable(activePipelines.get(pipelineName))
+				.map(savedPipeline -> {
+					activePipelines.remove(pipelineName);
+					try {
+						Files.delete(savedPipeline.file.toPath());
+						return true;
+					} catch (IOException e) {
+						return false;
+					}
+				}).ifPresentOrElse(ignored -> log.info("Pipeline {} was successfully removed and deleted", pipelineName), () -> {
+					throw new PipelineDoesNotExistsException(pipelineName);
```

Comment:
I don't like this, you don't see this behavior in spring repositories for mongo, postgres, etc. either.. When I call a delete on a repository, I do not expect an exception when it has already been deleted. The application will scale better and be more robust when it just does nothing when it is already deleted.